src/Orleans.Runtime/Hosting/Generic/SiloWrapper.cs

```diff
@@ -51,12 +51,21 @@ public async Task StopAsync(CancellationToken cancellationToken)
         }
 
         /// <inheritdoc />
-        public void Dispose()
+        public void Dispose() => this.DisposeAsync().AsTask().GetAwaiter().GetResult();
+
+        public async ValueTask DisposeAsync()
         {
             if (!isDisposing)
             {
                 this.isDisposing = true;
-                (this.Services as IDisposable)?.Dispose();
+                if (this.Services is IAsyncDisposable asyncDisposable)
+                {
+                    await asyncDisposable.DisposeAsync();
```

Comment:
is this the typical behavior (to preserve ordering)? or should you parallelize `DisposeAsync()` calls?

Reply:
There's only one call here - the important thing is to call DisposeAsync preferentially and only fall back to Dispose if async isn't supported